src/shared/components/common/markdown-textarea.tsx

```diff
@@ -374,10 +374,63 @@ export class MarkdownTextArea extends Component<
     autosize.update(textarea);
   }
 
-  handleImageUploadPaste(i: MarkdownTextArea, event: any) {
+  handlePaste(i: MarkdownTextArea, event: ClipboardEvent) {
+    if (!event.clipboardData) return;
+
+    // check clipboard files
     const image = event.clipboardData.files[0];
     if (image) {
       i.handleImageUpload(i, image);
+      return;
+    }
+
+    // check clipboard url
+    const url = i.isValidUrl(event.clipboardData.getData("text"));
+    if (url) {
+      i.handleUrlPaste(url, i, event);
+      return;
+    }
+  }
+
+  handleUrlPaste(url: URL, i: MarkdownTextArea, event: ClipboardEvent) {
+    // query textarea element
+    const textarea = document.getElementById(i.id);
+
+    if (textarea instanceof HTMLTextAreaElement) {
+      const { selectionStart, selectionEnd } = textarea;
+
+      // if no selection, just insert url
+      if (selectionStart === selectionEnd) return;
+
+      event.preventDefault();
+      const selectedText = i.getSelectedText();
+
+      // update textarea content
+      i.setState(({ content }) => ({
+        content: `${
+          content?.substring(0, selectionStart) ?? ""
+        }[${selectedText}](${url.toString()})${
+          content?.substring(selectionEnd) ?? ""
+        }`,
+      }));
+      i.contentChange();
+
+      // shift selection 1 to the right
+      textarea.setSelectionRange(
+        selectionStart + 1,
+        selectionStart + 1 + selectedText.length,
+      );
+    }
+  }
+
+  isValidUrl(value: string) {
+    if (!value) return false;
```

Comment:
This isn't necessary, you aren't passing in `string | undefined`, so we can assume its there.

Reply:
Due to the beauty of javascript and `""` being falsy, it does something :) It actually prevents a constructor call to `URL` if you pass in an empty string which would never be a valid URL. I'm moving to the `validUrl` helper function, so it won't be relevant.